app/src/main/java/com/andrew/tasky/presentation/reminder_detail/ReminderDetailFragment.kt

```diff
@@ -1,32 +1,341 @@
+
+            collectLatestLifecycleFlow(viewModel.selectedReminderTime) { reminderTime ->
+                when (reminderTime) {
+                    ReminderTime.TEN_MINUTES_BEFORE ->
+                        reminderLayout.reminderTextView.text =
+                            getString(R.string.ten_minutes_before)
+                    ReminderTime.THIRTY_MINUTES_BEFORE ->
+                        reminderLayout.reminderTextView.text =
+                            getString(R.string.thirty_minutes_before)
+                    ReminderTime.ONE_HOUR_BEFORE ->
+                        reminderLayout.reminderTextView.text =
+                            getString(R.string.one_hour_before)
+                    ReminderTime.SIX_HOURS_BEFORE ->
+                        reminderLayout.reminderTextView.text =
+                            getString(R.string.six_hours_before)
+                    ReminderTime.ONE_DAY_BEFORE ->
+                        reminderLayout.reminderTextView.text =
+                            getString(R.string.one_day_before)
+                }
+            }
+        }
+    }
+
+    private fun navigateToEditFragment(editType: EditType) {
+        setFragmentResultListener("INPUT_REQUEST_KEY") { resultKey, bundle ->
+            if (resultKey == "INPUT_REQUEST_KEY") {
+                val input = bundle.getString("INPUT")
+                when (editType) {
+                    EditType.TITLE -> {
+                        if (input != null) {
+                            viewModel.setTitle(input)
+                        }
+                    }
+                    EditType.DESCRIPTION -> {
+                        if (input != null) {
+                            viewModel.setDescription(input)
+                        }
+                    }
+                }
+            }
+        }
+        when (editType) {
+            EditType.TITLE -> {
+                val text = binding.addTitleAndDoneButtonLayout.titleTextView.text.toString()
+                val bundle = Bundle()
+                bundle.putString("TEXT", text)
+                bundle.putString("EDIT_TYPE", editType.name)
+
+                setFragmentResult("EDIT_TYPE_AND_TEXT_REQUEST_KEY", bundle)
+
+                navController.navigate(
+                    ReminderDetailFragmentDirections
+                        .actionReminderDetailFragmentToEditFragment()
+                )
+            }
+            EditType.DESCRIPTION -> {
+                val text = binding.addDescriptionLayout.descriptionTextView.text.toString()
+                val bundle = Bundle()
+                bundle.putString("TEXT", text)
+                bundle.putString("EDIT_TYPE", editType.name)
+
+                setFragmentResult("EDIT_TYPE_AND_TEXT_REQUEST_KEY", bundle)
+
+                navController.navigate(
+                    ReminderDetailFragmentDirections
+                        .actionReminderDetailFragmentToEditFragment()
+                )
+            }
+        }
```

Comment:
You could think about making this an extension function of `Fragment`, since you'll otherwise need to have this same code in all 3 types of fragments. The function signature could look something like this:
```
fun Fragment.navigateToEditFragment(editType: EditType, onResult: (String) -> Unit) {
    val navController = findNavController()
    // Put the same code in here
}
```

Call it like this, for example in `ReminderDetailFragment`:
```
navigateToEditFragment(
    editType = EditType.TITLE,
    onResult = viewModel::setTitle
)
```

PS: You'll also need to make the navigation to the edit fragment a global navigation, so you can call it from any fragment and not only those where you set a direct transition to it.